src/app.jsx

```diff
@@ -12,6 +13,55 @@ import routes from './routes';
 import analytics from './services/analytics';
 
 const App = () => {
+  // Code for setting up color schemes
```

Comment:
This code is, as you feared, cluttering App up. Please extract it a custom hook.

Reply:
After getting deep into reviewing this code, I realized that - I believe - most of it is unnecessary.

We don't need to watch for changes to the user's color scheme preference at the app level. If they change it, it will be because they clicked the toggle, so we can handle the change (AKA call `setMode` and update `localStorage`) within the toggle's callback.

So the only thing that we need to do when the app first renders is read the saved preference if it exists or else set it from the system preference. That only needs to happen once when the App mounts, no event listeners needed.

Reply:
Don't we want the color to update whenever the system setting changes if the user has selected system, without needing a reload?  That's why the listener is being used...

Reply:
Ah yes, that's what I was forgetting, thank you! Never mind then, we do need most of this.

Reply:
I won't be here the rest of the week, so I'm just gonna to leave here how I would write the hook, in semi-pseudocode.

We can actually dramatically simplify things if we just read the user's preference when the system color scheme changes, and only update if they are using the system theme. That way we don't need to listen for their preference, we just query at the time that it's relevant. And since the system color scheme doesn't change very often, this is probably actually the best way to do it.

```javascript
const useWatchUsersColorScheme = () => {
  const { setMode } = useColorScheme();

  useEffect(() => {
    const prefersDarkQuery = window.matchMedia('(prefers-color-scheme: dark)');
  
    const onSystemColorSchemeChange = (e) => {
      const colorSchemeSource = localStorage.getItem('colorMode') ?? 'system'
  
      if (colorSchemeSource === 'system') {
        setMode(e.matches ? 'dark' : 'light');
      }
    }
    
    prefersDarkQuery.addEventListener('change', onSystemColorSchemeChange);

    return () => {
      prefersDarkQuery.removeEventListener('change', onSystemColorSchemeChange);
    };
  }, [setMode]);
}
```

Reply:
Ah, gotcha, well, I'll see what I can do about simplifying it in the morning, thanks for all the help!

Reply:
I was able to simplify it a bunch, and extract it into a custom hook.  Along with a couple other use effects to ensure the color is correct when the app launches it seems to all work well, and I haven't encountered any issues with it from my testing.